api/users/views.py

```diff
@@ -290,7 +291,12 @@ def get_object(self):
         user_settings = self.get_addon_settings(check_object_permissions=False)
         account_id = self.kwargs['account_id']
 
-        account = ExternalAccount.load(account_id)
+        account = ExternalAccount.objects.filter(
+            _id=account_id,
+        ).annotate(
+            user_id=Value(user_settings.owner._id),
+        ).get()
```

Comment:
should be able to use the `external_accounts` property on the user_settings instead of reproducing the query. Maybe with a `first` instead of a `get` or do we want that to blow up if something unexpected happens?